param/__init__.py

```diff
@@ -870,6 +872,9 @@ def _validate(self, val):
         if not _is_number(val):
             raise ValueError("Parameter '%s' only takes numeric values"%(self.name))
 
+        if self.step is not None and not _is_number(self.step):
```

Comment:
Does setting a Parameter attribute actually trigger this validation? I'm not actually sure we implement that. Something for another PR of course but that seems like a pretty big hole in the validation now that it's really easy to set parameter attributes dynamically.

Reply:
That is a good point. It seems that having to validate in multiple places is pretty redundant from a code perspective - I'll now have a look to see where that is.